### ISSUE TITLE
Exact starts-with should take precidence

### ISSUE THREAD

**Issue by timscott** [atom/fuzzy-finder#41]
To replicate:
1. Open a tree containing files `cox.rb` and `context.rb`
2. `cmd-p` and type "cox".
3. Fuzzy finder lists `context.rb` above  `cox.rb`

Perhaps not _wrong_ but certainly [_surprising_](http://en.wikipedia.org/wiki/Principle_of_least_astonishment).

**Comment by bprosnitz** [comment]
+1 Exact searches sometimes don't match

**Comment by paulcsmith** [comment]
:+1:

**Comment by gdaolewe** [comment]
:+1: Large project with a lot of files, exact matches are often buried by results that have several of the search characters but spread far apart. This makes fuzzy finder almost less useful than if it weren't fuzzy at all.

**Comment by benzen** [comment]
I would argue that it's quite frequent to prefer matching at the end of the
file path than anywhere at the begining.
I mean the file name it almost always the only part of the path that i can
recall

2015-10-05 17:11 GMT-04:00 gdaolewe <notifications@github.com>:

> [image: :+1:] Large project with a lot of files, exact matches are often
> buried by results that have several of the search characters but spread far
> apart. This makes fuzzy finder almost less useful than if it weren't fuzzy
> at all.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/atom/fuzzy-finder/issues/41#issuecomment-145669703>.
>



-- 
Benjamin DREUX

**Comment by bronson** [comment]
Related to #98 no doubt.

@izuzak this feels buggy to me.  Not sure I agree with the enhancement label.

<img width="722" alt="screen shot 2015-10-10 at 4 12 29 pm" src="https://cloud.githubusercontent.com/assets/1776/10413750/c5bc926c-6f69-11e5-8122-8020e23ac1e0.png">

Yes I want job_spec.rb...  that's what I'm typing ain't it?  :)

**Comment by timscott** [comment]
@broson Yes, this is the exact case. Because of this, a simple starts-with search would be better like 80% of the time.

**Comment by gdaolewe** [comment]
@benzen I agree that it prefer matching the filename rather than the path, but it should prefer matching towards the beginning of the filename itself. If the matching is done on the full path + filename as one string, I think that's quite wrong.

I don't know how many I speak for, but I almost never type part of the path to find a file; the only reason I would is this exact scenario, where typing the filename exactly puts the desired file way down in the suggestions and I have to start typing the path to disambiguate.

**Comment by benzen** [comment]
Maybe I badly explained myself, but I totally agree with @gdaolewe
